service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageHearingsService.java

```diff
@@ -387,27 +415,55 @@ public HearingVenue getPreviousHearingVenue(CaseData caseData) {
         return hearingVenueLookUpService.getHearingVenue(previousHearingBooking);
     }
 
-    public Map<String, Object> populateFieldsWhenPastHearingDateAdded(LocalDateTime hearingStartDate,
-                                                                      LocalDateTime hearingEndDate) {
+    public Map<String, Object> populateFieldsWhenPastHearingDateAdded(CaseData caseData) {
+
         Map<String, Object> data = new HashMap<>();
         LocalDateTime currentDateTime = LocalDateTime.now();
 
         data.put(SHOW_PAST_HEARINGS_PAGE, NO.getValue());
 
-        if (hearingStartDate.isBefore(currentDateTime)) {
-            data.put(HEARING_START_DATE_LABEL, formatLocalDateTimeBaseUsingFormat(hearingStartDate, DATE_TIME));
+        if (caseData.getHearingStartDate().isBefore(currentDateTime)) {
+            data.put(HEARING_START_DATE_LABEL,
+                formatLocalDateTimeBaseUsingFormat(caseData.getHearingStartDate(), DateFormatterHelper.DATE_TIME));
             data.put(START_DATE_FLAG, YES.getValue());
             data.put(SHOW_PAST_HEARINGS_PAGE, YES.getValue());
         }
-        if (hearingEndDate.isBefore(currentDateTime)) {
-            data.put(HEARING_END_DATE_LABEL, formatLocalDateTimeBaseUsingFormat(hearingEndDate, DATE_TIME));
-            data.put(END_DATE_FLAG, YES.getValue());
-            data.put(SHOW_PAST_HEARINGS_PAGE, YES.getValue());
+
+        BiConsumer<LocalDateTime, String> populateFields = (endDateTime, endDateLabel) -> {
```

Comment:
I'd personally prefer this in a method instead of a BiConsumer.

Reply:
This is just to avoid passing all the variables in all the three conditions